source/gitlab/gitlab.go

```diff
@@ -113,13 +113,30 @@ func WithInstance(client *gitlab.Client, config *Config) (source.Driver, error)
 }
 
 func (g *Gitlab) readDirectory() error {
-	nodes, response, err := g.client.Repositories.ListTree(g.projectID, g.listOptions)
-	if err != nil {
-		return err
-	}
+	var (
+		nodes    []*gitlab.TreeNode
+		n        []*gitlab.TreeNode
+		response *gitlab.Response
+		err      error
+	)
+
+	for {
+		n, response, err = g.client.Repositories.ListTree(g.projectID, g.listOptions)
+		if err != nil {
+			return err
+		}
+
+		if response.StatusCode != http.StatusOK {
+			return ErrInvalidResponse
+		}
+
+		nodes = append(nodes, n...)
+
+		if response.NextPage == 0 {
```

Comment:
I don't see docs where `X-Next-Page` will be set to `0` when there are no pages left.

Insteasd, use `response.CurrentPage`. See [this example](https://github.com/xanzy/go-gitlab/blob/v0.31.0/examples/pagination.go#L36).

Other references:
* https://godoc.org/github.com/xanzy/go-gitlab#Response
* https://github.com/xanzy/go-gitlab/blob/v0.31.0/gitlab.go#L846